week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -39,12 +39,39 @@ DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
+var destinationNamesWithin500Kms = travelDestinations.filter(getNamesin500).map(getName);
 
-var destinationNamesWithin500Kms = // Complete here
+function getNamesin500(destination){
+    return (destination.distanceKms<500);
+}
 
-var destinationNameReachableByFerry = // Complete here
+function getName(destination){
+    return destination.destinationName;
+}    
 
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
+// Complete here
+
+var destinationNameReachableByFerry = travelDestinations.filter(getNamebyFerry).map(getName);
+
+function getNamebyFerry(destination){
+    return (destination.transportations.includes("ferry"));
+}
```

Comment:
As with `getNamesin500` above, `getNamebyFerry` does not return a *name*, so shouldn't be called `getName...`.  Think about what the function does, and think of a better name.  You don't need to consider how you use the function elsewhere, you only need to consider what the function *does* when it is called.